src/classes/ALLO_AllocationsDMLService.cls

```diff
@@ -0,0 +1,407 @@
+
+        if (dmlErrors) { // Need to log errors for all Opportunities that don't already have errors since retries exhausted
+            addErrorsWhenRetriesExceeded();
+        }
+
+        // Convert any errors that were not explicitly DML errors here to RecordErrors
+        processFunctionalErrors();
+    }
+
+    /**
+     * @description Clears any private member variables of data that needs to be reset between processing
+     */
+    public void reset() {
+        recordIdsWithErrors = new Set<Id>();
+        allRecordIdsWithDML = new Set<Id>();
+        allocationsWrapper = null;
+        processExclusions = null;
+    }
+
+    /**
+     * @description Processes Allocation deletions, recording DML errors as Record Errors and adding
+     * Record Exclusions for ones that errored.  The ALLO_AllocationsWrapper Allocations Maps are updated removing
+     * Allocations from the Maps for successful DML.
+     * @return Boolean, true if there were any DML errors
+     */
+    @testVisible private Boolean processAllocationDeleteResults() {
+        List<Database.DeleteResult> deletes = null;
+        boolean dmlErrorsPresent = false;
+
+        // Process Deletions
+        if (allocationsWrapper.allocationsForDelete != null && allocationsWrapper.allocationsForDelete.size() > 0) {
+            deletes = Database.delete(allocationsWrapper.allocationsForDelete, false);
+
+            // Handle Deletes and Update Maps and Errors
+            for (Integer i=0; i < deletes.size(); i++) {
+                Allocation__c deletedAlloc = allocationsWrapper.allocationsForDelete[i];
+                Database.DeleteResult delResult = deletes[i];
+                Id parentRecordId = (deletedAlloc.Opportunity__c != null) ? deletedAlloc.Opportunity__c : deletedAlloc.Payment__c;
+                allRecordIdsWithDML.add(parentRecordId);
+
+                if (delResult.isSuccess()) {
+                    // Removes the successfully deleted Allocation from the appropriate Allocations List
+                    allocationsWrapper.removeAllocation(deletedAlloc);
+                } else {
+                    dmlErrorsPresent = true;
+                    handleError(parentRecordId, deletedAlloc, delResult);
+                }
+            }
+        }
+
+        return dmlErrorsPresent;
+    }
+
+    /**
+     * @description Processes Allocation inserts, recording DML errors as Record Errors and adding
+     * Record Exclusions for ones that errored.  The ALLO_AllocationsWrapper Allocations Maps are updated adding all
+     * successful DML.
+     * @return Boolean true, if there were any dml errors
+     */
+    @testVisible private Boolean processAllocationInsertResults() {
+        List<Database.SaveResult> inserts = null;
+        Boolean dmlErrorsPresent = false;
+
+        // Process Inserts
+        if (allocationsWrapper.allocationsForInsert != null && allocationsWrapper.allocationsForInsert.size() > 0) {
```

Comment:
Same here with the check for a return followed an outdent.

Reply:
Done